zerver/tornado/event_queue.py

```diff
@@ -442,11 +442,19 @@ def add_client_gc_hook(hook: Callable[[int, ClientDescriptor, bool], None]) -> N
     gc_hooks.append(hook)
 
 
-def get_client_descriptor(queue_id: str) -> ClientDescriptor:
-    try:
-        return clients[queue_id]
-    except KeyError:
-        raise BadEventQueueIdError(queue_id)
+def get_client_descriptor(queue_id: str, user_profile_id: int) -> ClientDescriptor:
```

Comment:
Maybe call this `access_client_descriptor`, to follow our naming conventions for code doing access control checks?